src/backend/sgx/enclave.rs

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Enclave object
+
+use std::fs::{File, OpenOptions};
+use std::io::Result;
+
+pub const SGX_ENCLAVE_DEVICE: &str = "/dev/sgx_enclave";
+
+/// Wraps an enclave file descriptor.
+pub struct Enclave(File);
+
+impl From<Enclave> for File {
+    fn from(enclave: Enclave) -> File {
+        enclave.0
+    }
+}
+
+impl Enclave {
+    pub fn new() -> Result<Enclave> {
+        OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open(SGX_ENCLAVE_DEVICE)
+            .map(Self)
```

Comment:
neat :smile: ... never used a struct initializer with `map(Self)`

Reply:
Yeah, it was a random trial :-)